### DISCUSSION TITLE
How to create a slab phantom with a cd geometry?

### DISCUSSION THREAD

**Discussion by kstcmcjw** [nrc-cnrc/EGSnrc#879] [Questions] [answered]
Hello everyone, i am on line again for need some help, i write a geometry by myself, but can't simulation right, i carefully recheck the program, without any mistake, but i know there is something wrong.

The accident is this:  i meet a realistic problem, in the shielding projection, the X-ray room is shielded with lead plate for 1 to 3 mm thickness. Consider the cost-benefit model, the lead plate will spend a lot of money. So the compound shielding is birth. We usually use the concrete wall as the main structure and lead plate as the additional shielding. This method not only ensure the lead equivalent but also cut down the lead plate costs.

Then i simulation this scence, i use a EGS_plane and EGS_cdgeometry to creat two wall: the lead wall and concrete wall, actully, there are two plane, but i only simulated the lead plane, where the concrete plane?

My program as show:

```
:start geometry definition:
  :start geometry: 
    name = barrier_planes
    library = egs_planes
    type = EGS_Xplanes
    positions = -24.2 -0.2 0
  :stop geometry:

  :start geometry:
    name = barrier_lead
    library = egs_box
    box size = 0.2 100 100 
    midpoint = -0.1 0 0
        :start media input:
            media = lead
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_lead

  :start geometry:
    name = barrier_concrete
    library = egs_box
    box size =  24 100 100 
    midpoint = -12.2 0 0
        :start media input:
            media = concrete
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_concrete

  :start geometry:
    name = barrier
    library = egs_cdgeometry
    base geometry =barrier_planes
    set geometry = 1 barrier_lead
    set geometry = 0 barrier_concrete
  :stop geometry:
  simulation geometry = barrier

:stop geometry definition:
```

The EGS view software is show as:
![image](https://user-images.githubusercontent.com/96934217/172535950-dbc2ef9b-4e21-4f36-ab48-b46d7157d852.png)

I know something is wrong, where is the mistake?  Are the EGS_plane and EGS_cdgeometry not right?  Please help me ! 

Thanks for your attention! Wish you has a good mood! Thank you!

**Comment by ftessier** [comment, marked as answer]
Try this:

```
:start geometry definition:
  :start geometry: 
    name = barrier_planes
    library = egs_planes
    type = EGS_Xplanes
    positions = -24.2 -0.2 0
  :stop geometry:

  :start geometry:
    name = barrier_lead
    library = egs_box
    box size = 100 100 100       ### Make the box very large along x, without a midpoint: the cd will "cut" it with barrier planes
        :start media input:
            media = lead
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_lead

  :start geometry:
    name = barrier_concrete
    library = egs_box
    box size =  100 100 100      ### Make the box very large along x, without a midpoint: the cd will "cut" it with barrier planes
        :start media input:
            media = concrete
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_concrete

  :start geometry:
    name = barrier
    library = egs_cdgeometry
    base geometry = barrier_planes
    set geometry = 1 barrier_lead
    set geometry = 0 barrier_concrete
  :stop geometry:
  simulation geometry = barrier

:stop geometry definition:

**Reply by kstcmcjw** [reply]
er, thank you very much, i try it with your method, it show the simulation, and i go on my program, in the EGSnrc GUI software, completed with the Compile and Execute, but Execute failed.

it shows:

![image](https://user-images.githubusercontent.com/96934217/172830918-8ef5c38b-f205-4a00-8814-626377103db1.png)

i recheck my program, part of source:

```
###4.SOURCE
:start source definition:
    :start source:
        library = egs_collimated_source
        name = tube_source
        :start source shape:
            type = point
            position = 2000 0 0
        :stop source shape:
        :start target shape:
            Library = egs_rectangle
            rectangle = -200 -200 200 200
        :stop target shape:
            position = 0 0 0 
            distance = 2000
            charge = 0
        :start spectrum:
            type = monoenergetic
            energy = 0.12
        :stop spectrum:
    :stop source:
    simulation source = tube_source
:stop source definition:
```

it simulated the X-ray tube, from a point to an area of rectangle, 200mm distance from the lead wall, the target area is an rectangle, 20cm×20cm, but i can't find where the mistake is, whether the direction is to the positive of the X axis?  Far away from my geometry?

The whole program as the shows:

```
###1.RUN CONTROL
:start run control:
    ncase = 1e2 
:stop run control:

###2.GEOMETRY
:start geometry definition:
  :start geometry: 
    name = barrier_planes
    library = egs_planes
    type = EGS_Xplanes
    positions = -242 -2 0
  :stop geometry:

  :start geometry:
    name = barrier_lead
    library = egs_box
    box size = 1000 1000 1000
        :start media input:
            media = lead
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_lead

  :start geometry:
    name = barrier_concrete
    library = egs_box
    box size =  1000 1000 1000
        :start media input:
            media = concrete
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_concrete

  :start geometry:
    name = barrier
    library = egs_cdgeometry
    base geometry =barrier_planes
    set geometry = 1 barrier_lead
    set geometry = 0 barrier_concrete
  :stop geometry:
  simulation geometry = barrier

:stop geometry definition:

###3.MEDIA
:start media definition: 
  ae = 0.521 # lowest energy for electrons (kinetic+0.511)
  ap = 0.01 # lowest energy for photons (kinetic)
  ue = 50.511 # maximum energy for electrons (kinetic+0.511)
  up = 50 # maximum energy for photons (kinetic)
 
  :start concrete: # this name can be anything
      density correction file = concrete_ordinary_type03  # name the file (no ext.) 
  :stop concrete:
   
  :start lead:
      density correction file = lead
  :stop lead:
:stop media definition:

###4.SOURCE
:start source definition:
    :start source:
        library = egs_collimated_source
        name = tube_source
        :start source shape:
            type = point
            position = 2000 0 0
        :stop source shape:
        :start target shape:
            Library = egs_rectangle
            rectangle = -200 -200 200 200
        :stop target shape:
            distance = 2000
            charge = 0
        :start spectrum:
            type = monoenergetic
            energy = 0.12
        :stop spectrum:
    :stop source:
    simulation source = tube_source
:stop source definition:

###5.VIEWER CONTROL
:start view control:
  set color = lead 120 120 200 200
  set color = concrete 120 255 120 255
:stop view control:

###6.AUSGAB OBJECTS
:start ausgab object definition: 
    :start ausgab object:
      name = tracks
      library = egs_track_scoring
    :stop ausgab object:
:stop ausgab object definition:

:start ausgab object definition:
  :start ausgab object:
      name = dose
      library = egs_dose_scoring
      dose regions = -542 0 0
      volume = 10
  :stop ausgab object:
:stop ausgab object definition:
```

but failed,  it shows:  `1EGS_CollimatedSource::getPositionDirection:   my target shape rectangle_7, which is of type rectangle, failed to   return a positive weight after 10000 attempts`

why? where is the mistake? Thanks for your help!

**Comment by ftessier** [comment]
The rectangle shape by definition is in the `xy` plane (normal vector along z-axis), and your source shape is a point on the x-axis, so the "cross section" of the target shape as viewed from the source point is null, which explains that emitted particles all have `weight = 0.`

The solution is to position the source along the z-axis, or else rotate the rectangle so that its normal is parallel to the x-axis, with a `:start transformation: (...) :stop transformation:` block (see the [Affine Transform egs++ class](https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__AffineTransform.html#a601a07115aecff1925956a9097e69950)):

```:
start target shape:
    library = egs_rectangle
    rectangle = -200 -200 200 200
    :start transformation:
        rotation = 0 1.57079632679 0    ### rotate 90 degrees around the y-axis
    :stop transformation:
:stop target shape:
```

**Reply by ftessier** [reply]
In general it will be easier to setup your geometry so that the beam direction is along the z-axis, since this is the "default" orientation for many objects in egs++ (for example, the rectangle shape is in the `xy` plane). Using another axis for the beam is perfectly fine, but you will likely have to perform rotations here and there.

**Comment by ftessier** [comment]
With the proposed change (changed lead to air in order to view photon tracks through the box!):

![image](https://user-images.githubusercontent.com/1444241/172850226-2b1351f3-61fb-4864-aa2a-dfa9e6f14d1e.png)

**Reply by kstcmcjw** [reply]
oh, it is, i am apologzise that i did not consider this situation, yes, X-ray generate is a point, and the ray though the light-jaw, and a quartet cone has accomplished. My mistake should consider from one point to the two-dimensional surface.

Thank you very much! By the way, i want know one probloms: is in the `$EGSnrc/HEN_HOUSE/pgs4/density_corrections/compounds`, I can't find the brick. 

Why i what the brick? Because in our radiation protection project, we use the brick-wall as the main strcture, and the lead plate on the brick wall.

-----------------------------------------------------------

The scene as the show:

![image](https://user-images.githubusercontent.com/96934217/172854218-80187762-7260-4321-8871-022722e8ad39.png)

![image](https://user-images.githubusercontent.com/96934217/172854833-9495e6d8-75a6-44a0-88fd-c237f9747ea8.png)

**Reply by ftessier** [reply]
Yes, first you need to find the composition of your birck (bulk density and chemical composition). Even the concrete(s) supplied with EGSnrc might not be appropriate; there are many different types and compositions (there are 8 supplied with EGSnrc). Once you have material density and composition, you can create your own density correction file which defines the medium: see the [instructions to define a material](https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR) in the EGSnrc wiki.

**Reply by ftessier** [reply]
By the way, in your modelling you have to worry about spaces between the bricks, and the composition of the mortar, if significantly different than that of the bricks. Tricky business!

**Reply by kstcmcjw** [reply]
It is very very kind of you to response my so many questions, thank you very much! 
I love radiation protection, from three principles of protection to radiation protection programme, from technology to manegment, i do love it, so i have many questions, before i contect EGS, i study the basics of radiation protection, so many equations. Some of the equation don't suit to the practice scene, for example the caculation of electron, caculation of photon though the compound protection barrier wall.

EGSnrc is really a precious deposits, i will study it very hardly!

Thank you very much !!!

**Reply by kstcmcjw** [reply]
er, i am back with some progress, i modified my program, so that linking theory with practice, and simulated again, there are 7 problems i want to discuss and consult with an open mind.
 Well, first of all, i bring my program here:

--------------------------------------------------------------------------------------------------------------------------------------------------------

###1.RUN CONTROL
:start run control:
    ncase = 1e5
:stop run control:

###2.GEOMETRY
:start geometry definition:
  :start geometry: 
    name = cd_planes
    library = egs_planes
    type = EGS_Zplanes
    positions = -600 -242 -2 0 3000
  :stop geometry:

  :start geometry:
    name = air0
    library = egs_box
    box size =  1000 1000 3000
        :start media input:
            media = air
        :stop media input:
  :stop geometry:
  simulation geometry = air0

  :start geometry:
    name = barrier_lead
    library = egs_box
    box size = 1000 1000 1000
        :start media input:
            media = lead
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_lead

 :start geometry:
    name = barrier_concrete
    library = egs_box
    box size = 1000 1000 1000
        :start media input:
            media = concrete
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_concrete

  :start geometry:
    name = air1
    library = egs_box
    box size =  1000 1000 1000
        :start media input:
            media = air
        :stop media input:
  :stop geometry:
  simulation geometry = air1
  
  :start geometry:
    name = protection
    library = egs_cdgeometry
    base geometry =cd_planes
    set geometry = 3 air0
    set geometry = 2 barrier_lead
    set geometry = 1 barrier_concrete
    set geometry = 0 air1
  :stop geometry:
  simulation geometry = protection

:stop geometry definition:

###3.MEDIA
:start media definition: 
  ae = 0.521 # lowest energy for electrons (kinetic+0.511)
  ap = 0.01 # lowest energy for photons (kinetic)
  ue = 50.511 # maximum energy for electrons (kinetic+0.511)
  up = 50 # maximum energy for photons (kinetic)
 
  :start concrete: # this name can be anything
      density correction file = concrete_ordinary_type03  # name the file (no ext.) 
  :stop concrete:
   
  :start lead:
      density correction file = lead
  :stop lead:

  :start air:
      density correction file = air_dry_nearsealevel
  :stop air:

:stop media definition:

###4.SOURCE
:start source definition:
    :start source:
        library = egs_collimated_source
        name = tube_source
        :start source shape:
            type = point
            position = 0 0 2000
        :stop source shape:
        :start target shape:
            library = egs_rectangle
            rectangle = -200 -200 200 200
        :stop target shape:
            distance = 2000
            charge = 0
        :start spectrum:
            type = monoenergetic
            energy = 0.12
        :stop spectrum:
    :stop source:
    simulation source = tube_source
:stop source definition:

###5.VIEWER CONTROL
:start view control:
  set color = lead 160 65 105 255
  set color = concrete 120 255 120 255
  set color = air0 0 200 50 50
  set color = air1 0 200 50 50
:stop view control:

###6.AUSGAB OBJECTS
:start ausgab object definition: 
    :start ausgab object:
      name = tracks
      library = egs_track_scoring
    :stop ausgab object:
:stop ausgab object definition:

:start ausgab object definition:
  :start ausgab object:
      name = dose
      library = egs_dose_scoring
      dose regions = 0
      volume = 10
  :stop ausgab object:
:stop ausgab object definition:

--------------------------------------------------------------------------------------------------------------------------------------------------------

compiled and executed it, it shows the figure as the show:

![image](https://user-images.githubusercontent.com/96934217/173224937-afa0e6ee-dd92-4d3a-a4f4-7ed44d743d68.png)

but the figure is out of my expectation, i want to express the geometry is from a point source generate, and through the air to the main barrier(2mm Pb plate and 24cm concrete), a rectangle area expoured on the barrier, and there set a monitor point out of the main barrier, 30 cm distance from the concrete, then through the EGSnrc software, i can get the dose and the dose-rate in the monitor point, my thought is the figure as the show:


![无标题](https://user-images.githubusercontent.com/96934217/173226734-77f936dd-e82b-4b2f-86f5-7561b77cb563.jpg)

like the example: the CollimatedSource :

![ccdc7a212f73241d64d9b664c8c456a9_egs_collimated_source](https://user-images.githubusercontent.com/96934217/173227015-e635b100-8578-4601-8d33-3a94674bc223.png)

that is all, my problems is as the follow:

1. I pozzled with the "library" and "type", i know the library, but i can't find the type description, what's the difference?  And what is the type significance.
2. Some source library don't have the "direction",for example: direction = 0 0 1, whether the source direction is to the negative Z-axie all the time as the default. 
3. In the EGS_DoseScoring Class Reference, The ausgab object can be used to calculate dose in identical voxels during run time and to output this information at the end of the calculation:

:start ausgab object:
  library      = egs_dose_scoring
  name         = some_name
  medium dose  = yes # no  (default)
  region dose  = no  # yes (default)
  volume       =  v1 ... vn  # in cm**3
  dose regions = ir1 ... irn # individual regions
:stop ausgab object:

in the ausgab object, what the difference between medium dose and region dose? I think the two is one thing? And the volume is the independence area? Whether dose regions is to the volume for one by one?

5. In my program, how to simulate the dose region? Should i add a chamber?
6. The color of air, EGS_VIEW display red air from the beginning to the end, regardless i modified it. Why?
7. How to simulate the dose-rate after shielding，Should i define the output dose-rate in the particle source ?

That is my all problem, i want use the EGSnrc to solve the practice problem in the radiation shielding protection project, and i feel the EGS software is very suit for me, thanks for you attention, wish you have a nice mood everyday! 

Thank you!

**Comment by kstcmcjw** [comment]
er, i am back with some progress, i modified my program, so that linking theory with practice, and simulated again, there are 7 problems i want to discuss and consult with an open mind.
Well, first of all, i bring my program here:

##https://github.com/nrc-cnrc/EGSnrc/pull/1.RUN CONTROL
:start run control:
ncase = 1e5
:stop run control:

##https://github.com/nrc-cnrc/EGSnrc/issues/2.GEOMETRY
:start geometry definition:
:start geometry:
name = cd_planes
library = egs_planes
type = EGS_Zplanes
positions = -600 -242 -2 0 3000
:stop geometry:

:start geometry:
name = air0
library = egs_box
box size = 1000 1000 3000
:start media input:
media = air
:stop media input:
:stop geometry:
simulation geometry = air0

:start geometry:
name = barrier_lead
library = egs_box
box size = 1000 1000 1000
:start media input:
media = lead
:stop media input:
:stop geometry:
simulation geometry = barrier_lead

:start geometry:
name = barrier_concrete
library = egs_box
box size = 1000 1000 1000
:start media input:
media = concrete
:stop media input:
:stop geometry:
simulation geometry = barrier_concrete

:start geometry:
name = air1
library = egs_box
box size = 1000 1000 1000
:start media input:
media = air
:stop media input:
:stop geometry:
simulation geometry = air1

:start geometry:
name = protection
library = egs_cdgeometry
base geometry =cd_planes
set geometry = 3 air0
set geometry = 2 barrier_lead
set geometry = 1 barrier_concrete
set geometry = 0 air1
:stop geometry:
simulation geometry = protection

:stop geometry definition:

##https://github.com/nrc-cnrc/EGSnrc/issues/3.MEDIA
:start media definition:
ae = 0.521 # lowest energy for electrons (kinetic+0.511)
ap = 0.01 # lowest energy for photons (kinetic)
ue = 50.511 # maximum energy for electrons (kinetic+0.511)
up = 50 # maximum energy for photons (kinetic)

:start concrete: # this name can be anything
density correction file = concrete_ordinary_type03 # name the file (no ext.)
:stop concrete:

:start lead:
density correction file = lead
:stop lead:

:start air:
density correction file = air_dry_nearsealevel
:stop air:

:stop media definition:

##https://github.com/nrc-cnrc/EGSnrc/issues/4.SOURCE
:start source definition:
:start source:
library = egs_collimated_source
name = tube_source
:start source shape:
type = point
position = 0 0 2000
:stop source shape:
:start target shape:
library = egs_rectangle
rectangle = -200 -200 200 200
:stop target shape:
distance = 2000
charge = 0
:start spectrum:
type = monoenergetic
energy = 0.12
:stop spectrum:
:stop source:
simulation source = tube_source
:stop source definition:

##https://github.com/nrc-cnrc/EGSnrc/issues/5.VIEWER CONTROL
:start view control:
set color = lead 160 65 105 255
set color = concrete 120 255 120 255
set color = air0 0 200 50 50
set color = air1 0 200 50 50
:stop view control:

##https://github.com/nrc-cnrc/EGSnrc/issues/6.AUSGAB OBJECTS
:start ausgab object definition:
:start ausgab object:
name = tracks
library = egs_track_scoring
:stop ausgab object:
:stop ausgab object definition:

:start ausgab object definition:
:start ausgab object:
name = dose
library = egs_dose_scoring
dose regions = 0
volume = 10
:stop ausgab object:
:stop ausgab object definition:

compiled and executed it, it shows the figure as the show:

![173224937-afa0e6ee-dd92-4d3a-a4f4-7ed44d743d68](https://user-images.githubusercontent.com/96934217/173271244-e9f07e41-9c0e-49db-8f7c-f84a300a3403.png)


but the figure is out of my expectation, i want to express the geometry is from a point source generate, and through the air to the main barrier(2mm Pb plate and 24cm concrete), a rectangle area expoured on the barrier, and there set a monitor point out of the main barrier, 30 cm distance from the concrete, then through the EGSnrc software, i can get the dose and the dose-rate in the monitor point, my thought is the figure as the show:

![173226734-77f936dd-e82b-4b2f-86f5-7561b77cb563](https://user-images.githubusercontent.com/96934217/173271259-49ab024a-c066-4a64-addf-2b3d32219d90.jpg)


like the example: the CollimatedSource :

![173227015-e635b100-8578-4601-8d33-3a94674bc223](https://user-images.githubusercontent.com/96934217/173271288-c5e86395-eca6-4137-95a5-0503d29a8fd5.png)


that is all, my problems is as the follow:

I pozzled with the "library" and "type", i know the library, but i can't find the type description, what's the difference? And what is the type significance.
Some source library don't have the "direction",for example: direction = 0 0 1, whether the source direction is to the negative Z-axie all the time as the default.
In the EGS_DoseScoring Class Reference, The ausgab object can be used to calculate dose in identical voxels during run time and to output this information at the end of the calculation:
:start ausgab object:
library = egs_dose_scoring
name = some_name
medium dose = yes # no (default)
region dose = no # yes (default)
volume = v1 ... vn # in cm**3
dose regions = ir1 ... irn # individual regions
:stop ausgab object:

in the ausgab object, what the difference between medium dose and region dose? I think the two is one thing? And the volume is the independence area? Whether dose regions is to the volume for one by one?

In my program, how to simulate the dose region? Should i add a chamber?
The color of air, EGS_VIEW display red air from the beginning to the end, regardless i modified it. Why?
How to simulate the dose-rate after shielding，Should i define the output dose-rate in the particle source ?
That is my all problem, i want use the EGSnrc to solve the practice problem in the radiation shielding protection project, and i feel the EGS software is very suit for me, thanks for you attention, wish you have a nice mood everyday!

Thank you!

**Reply by rtownson** [reply]
```
:start source shape:
type = point
position = 0 0 2000
:stop source shape:
```

It looks like you are expecting the distance to be in millimeters, because I think you were aiming for 2m and here you have 20m. **All distance units in the egsinp file are in `cm` and all energies are in `MeV`.**

> I pozzled with the "library" and "type"

Just use what you find in the documentation, they are essentially the same. There's a technical reason why these are two different things but it's not important to understand.

> Some source library don't have the "direction"

That's right. In those cases you have to use a source transformation.
https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__TransformedSource.html

> in the ausgab object, what the difference between medium dose and region dose
This is described in the documentation

https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__DoseScoring.html
Region dose will return the dose by region, medium dose is by medium (all regions combined for each medium). The volume you must calculate yourself and provide in order to get the right dose. The energy deposited will still be correct if you don't know the volumes. Use 1 number to set all volumes the same, otherwise list them in the same order you list `dose regions`.

> In my program, how to simulate the dose region

Not sure what you mean. If you want dose in a chamber, add a chamber. If you want dose in some other material, just find the region number corresponding to that volume and list it in 'dose regions' and it will print the energy/dose deposited.

> The color of air

You set it for `air0` and `air1` but your material is named `air`.

> How to simulate the dose-rate

The MC simulation does not model time. To get dose rate you would need to calibrate the MC simulation with experiment, for the source you are modelling. It's usually better to consider relative comparisons with MC than seek absolute dose.

**Reply by kstcmcjw** [reply]
First of all, it is very kind of you to answer my question, and your guidance are very particular! thank you very much, er, other questions i have clearness, but only one question i can't understand, it is the shape of the source simulation.

![image](https://user-images.githubusercontent.com/96934217/174705542-7b9a8d6c-177b-4e8f-b37a-3d82bb713bfb.png)


As you see, the X-ray generator and the X-ray's target shape is rectangle is regular cubic cone(like the pyramid shape), and I want to simulate this shape, but why i simulated shape is trapezoid cone, it seems may lack of the top pyramid shape, and i tried once more time, can't simulated the pyramid shape, i really do not know why?

I want to simulate the DR's beam, er, a simple simulation, i want to simulate the tube as a point source, then the target area is a rectangle 

![image](https://user-images.githubusercontent.com/96934217/174705874-2a022e01-c989-4904-b71b-a86b39e82cee.png)

but in my EGSnrc program it can't simulate the shape of pyramid, a trapezoid cone shape instead!

that is my code:

:start source definition:
:start source:
library = egs_collimated_source
name = tube_source
:start source shape:
type = point
position = 0 0 2000
:stop source shape:
:start target shape:
library = egs_rectangle
rectangle = -200 -200 200 200
:stop target shape:
distance = 2000
charge = 0
:start spectrum:
type = monoenergetic
energy = 0.12
:stop spectrum:
:stop source:
simulation source = tube_source
:stop source definition:

**Reply by rtownson** [reply]
Your air box is 3000cm along the z-axis, centered on the origin. So it goes -1500 to +1500. But your source is positioned at +2000cm, so the point source is positioned outside the air box and appears cut off when you display the geometry. Just make your air box larger. Also, remember that these dimensions are in cm, so I think much larger than you intend.

**Reply by kstcmcjw** [reply]
Thank you very much for your guidance, yes, the problem is in my geometry, I ignored the midpoint of EGS_box, and it happened that the program introduction did not explain where the midpoint is, so I ignored it, resulting in a clipping plane smaller than the source beam, and the beam "missing pyramid shape" occurs.

![13837a7831dd43c5dd722cafb0f67ed](https://user-images.githubusercontent.com/96934217/174828715-8d2b2fa7-6dba-4542-9e5b-f3c5091fd01f.png)
I have accomplish this simulation.

Thank you very much and thanks ftessier for your and his answers, my next goal is to refine your guidance one by one, and then use EGSnrc to improve this practical problem
Thank you very much and have a nice day!
